subprojects/plugins/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -218,7 +219,14 @@ private Configuration createApiElements(SourceSet sourceSet, PublishArtifact jar
     }
 
     private Configuration createSourceElements(ConfigurationContainer configurations, ProviderFactory providerFactory, ObjectFactory objectFactory, SourceSet sourceSet) {
-        Configuration variant = configurations.create(SOURCE_ELEMENTS_VARIANT_NAME);
+
+        // TODO: Why are we using this non-standard name? For the `java` component, this
+        // equates to `mainSourceElements` instead of `sourceElements` as one would expect.
+        // Can we change this name without breaking compatibility? Is the variant name part
+        // of the component's API?
+        String variantName = sourceSet.getName() + SOURCE_ELEMENTS_VARIANT_NAME_SUFFIX;
```

Comment:
👍 This is a good question.  Have we ever removed a consumable variant created by a core plugin before?  What is the deprecation procedure for variants?  We're just changing a configuration name here, but it seems like we would have go through a deprecation cycle.

Reply:
Leaving this TODO in for now as it does not need to be solved in this PR